src/Compilers/CSharp/Test/Emit/Emit/EmitMetadataTests.cs

```diff
@@ -3175,6 +3175,27 @@ .maxstack  1
                     """);
         }
 
+        [Fact]
+        public void DataSectionStringLiterals_HashCollision()
+        {
+            var emitOptions = new EmitOptions
+            {
+                // Take only the first byte of each string as its hash to simulate collisions.
+                TestOnly_DataToHexViaXxHash128 = static (data) => data[0].ToString(),
+            };
+            var source = """
+                System.Console.Write("a");
+                System.Console.Write("b");
+                System.Console.Write("aa");
+                """;
+            CreateCompilation(source,
+                parseOptions: TestOptions.Regular.WithFeature("experimental-data-section-string-literals", "0"))
+                .VerifyEmitDiagnostics(emitOptions,
+                    // (3,22): error CS9274: Cannot emit this string literal into the data section because it has XXHash128 collision with another string literal: a
+                    // System.Console.Write("aa");
+                    Diagnostic(ErrorCode.ERR_DataSectionStringLiteralHashCollision, @"""aa""").WithArguments("a").WithLocation(3, 22));
```

Comment:
>ERR_DataSectionStringLiteralHashCollision [](http://example.com/codeflow?start=41&length=41)

How should I user work around this issue if they hit it? By disabling the feature?

Reply:
Yes, or changing the threshold to exclude one of the conflicting strings.

Or they should go buy a lottery ticket since they seem to be very lucky to hit a xxh128 collision :-D